File: problems/0128-longest-consecutive-sequence/go/solution.go
package solution

// Longest Consecutive Sequence — https://leetcode.com/problems/longest-consecutive-sequence/
func longestConsecutive(nums []int) int {
	set := make(map[int]struct{}, len(nums))
	for _, n := range nums {
		set[n] = struct{}{}
	}
	longest := 0
	for n := range set {
		if _, ok := set[n-1]; ok {
			continue
		}
		length := 1
		for {
			if _, ok := set[n+length]; !ok {
				break
			}
			length++
		}
		if length > longest {
			longest = length
		}
	}
	return longest
}

File: problems/0128-longest-consecutive-sequence/typescript/solution.ts
// Longest Consecutive Sequence — https://leetcode.com/problems/longest-consecutive-sequence/
export function longestConsecutive(nums: number[]): number {
  const set = new Set(nums);
  let longest = 0;
  for (const n of set) {
    if (set.has(n - 1)) continue;
    let length = 1;
    while (set.has(n + length)) length++;
    longest = Math.max(longest, length);
  }
  return longest;
}
